Sources/Core/AWSClientRuntime/AWSClientConfigDefaultsProvider.swift

```diff
@@ -0,0 +1,105 @@
+
+    public static func credentialsProvider(
+        _ credentialsProvider: AWSClientRuntime.CredentialsProviding? = nil) throws -> CredentialsProviding {
+        let resolvedCredentialsProvider: AWSClientRuntime.CredentialsProviding
+        let fileBasedConfig = try CRTFileBasedConfiguration.make()
+        if let credentialsProvider = credentialsProvider {
+            resolvedCredentialsProvider = credentialsProvider
+        } else {
+            resolvedCredentialsProvider = try DefaultChainCredentialsProvider(fileBasedConfig: fileBasedConfig)
+        }
+        return resolvedCredentialsProvider
+    }
+
+    public static func region(_ region: String? = nil) async throws -> String? {
+        let resolvedRegion: String?
+        let fileBasedConfig = try await CRTFileBasedConfiguration.makeAsync()
+        if let region = region {
+            resolvedRegion = region
+        } else {
+            let regionResolver = try DefaultRegionResolver { _, _ in fileBasedConfig }
+            resolvedRegion = await regionResolver.getRegion()
+        }
+
+        return resolvedRegion
+    }
+
+    public static func appID(_ appID: String? = nil) throws -> String? {
+        let fileBasedConfig = try CRTFileBasedConfiguration.make()
+        let resolvedAppID: String?
+        if let appID = appID {
+            resolvedAppID = appID
+        } else {
+            resolvedAppID = AppIDConfig.appID(
+                configValue: nil,
+                profileName: nil,
+                fileBasedConfig: fileBasedConfig
+            )
+        }
+        return resolvedAppID
+    }
+
+    public static func retryMode(_ retryMode: AWSRetryMode? = nil) throws -> AWSRetryMode {
+        let fileBasedConfig = try CRTFileBasedConfiguration.make()
+        let resolvedRetryMode: AWSRetryMode?
+        if let retryMode = retryMode {
+            resolvedRetryMode = retryMode
+        } else {
+            resolvedRetryMode = AWSRetryConfig.retryMode(
+                configValue: nil,
+                profileName: nil,
+                fileBasedConfig: fileBasedConfig
+            )
+        }
+        return resolvedRetryMode!
+    }
+
+    public static func retryStrategyOptions() throws -> RetryStrategyOptions {
+        let fileBasedConfig = try CRTFileBasedConfiguration.make()
+        let resolvedMaxAttempts = AWSRetryConfig.maxAttempts(
+            configValue: nil,
+            profileName: nil,
+            fileBasedConfig: fileBasedConfig
+        )
+
+        let resolvedRateLimitingMode: RetryStrategyOptions.RateLimitingMode
+
+        // swiftlint:disable switch_case_alignment
+        switch try self.retryMode(nil) {
+            case .legacy, .standard:
+                resolvedRateLimitingMode = .standard
+            case .adaptive:
+                resolvedRateLimitingMode = .adaptive
+        }
+        // swiftlint:enable switch_case_alignment
+
```

Comment:
nit: should remove switftlint disable switch_case_alignment and move case statements over so that they are in-line with switch